### ISSUE TITLE
By any chance, could this mod be connected to this crash that's been plaguing my server?

### ISSUE THREAD

**Issue by Sunconure11** [OreCruncher/TidyChunk#1]
https://pastebin.com/hbN4kpYa

**Comment by OreCruncher** [comment]
I don't think so.  During generation TidyChunk captures the ChunkPos and stores it in the list.  The actual removal of the EnityItems occurs at potentially 3 different times:
* When the EntityItem joins the world.  If it's in the list of ChunkPos that TidyChunk has it cancels the event.
* When the world unloads.  TidyChunk makes a final pass of the entities looking for EntityItems that are in the ChunkPos list.
* During the end phase of the world tick.  Similar to when the world unloads.

If anything I would have expected some weird comodification from my ChunkPos list.

**Comment by Sunconure11** [comment]
https://www.dropbox.com/s/koqvocwqqrhscmz/logsforcrash.zip?dl=0

Here are various logs, relating to this. I'm trying to figure out WTF is causing this

**Comment by OreCruncher** [comment]
What stood out to me was from the log.  Thaum was doing it's thing with each of the worlds along with astral sorcery.  Looks like things were fine up until it hit dim 0.  Right after astral sorcery did it's thing the crash happened.  It could be the other dimensions were fine because there were no chunks being loaded.  For dim 0 server start I think it would load chunks especially around spawn.  So it seems the main issue is around dim 0 load, and I would look at astral sorcery first since it was the last thing reported in the log before the event.

(Oh and to satisfy curiosity you can remove TidyChunk from the mix just be sure.  I don't think its a problem, but given the quantity of mods in the pack I can't rule out interactions.  Is this version of the pack you are running on Twitch?)

**Comment by Sunconure11** [comment]
4.10.6 is the build of the pack, but it occurred in a few earlier builds before that one as well. I have not tested the build I just pushed out yet, to see if the error still occurs.

I have only seen it happen on servers, so far, not on singleplayer. Either that, or no one has reported this issue on singleplayer to me.

**Comment by Sunconure11** [comment]
https://www.dropbox.com/s/l7v6dy305edst2j/corrupted_world.zip?dl=0

Discussion with others has led to the conclusion of something corrupting the world. This is the world used on said server.

**Comment by OreCruncher** [comment]
Right now I have Forge 2768 in my dev environments (your server has 2772).  I looked at the code for 2768 base on the call stack and I am not sure how TidyChunk could cross paths with this.  Now, what I find scary is that in the Forge code there is a comment that says "//FML move above remove to prevent CMEs".  Makes me wonder if you stumbled across another corner case of something.  I will see about update one of my sandboxes to 2772 to see what it has to say.

EDIT: Code looks to be identical in these areas between 2768 and 2772.  Shot in the dark, but the pack was running fine up to a point, then something broke.  Do you recall anything you may have did around that time (update mods, etc.)?

**Comment by HellFirePvP** [comment]
//FML move above remove to prevent CMEs

I do wonder where you found that - skimmed through the forge commits and couldn't find it.

**Comment by OreCruncher** [comment]
@HellFirePvP I found it as a comment within the Forge JAR in my project - World.class.  Seems there is some sort of hackery related to getting TEs loaded in the world.

**Comment by Sunconure11** [comment]
I had pushed out several updates to my pack, but I'm not sure where it first appeared
